src/cliffordsynthesis/TargetMetricHandler.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+* This file is part of the MQT QMAP library which is released under the MIT license.
+* See file README.md or go to https://www.cda.cit.tum.de/research/ibm_qx_mapping/ for more information.
+*/
+
+#include "cliffordsynthesis/TargetMetricHandler.hpp"
+
+#include "cliffordsynthesis/GateEncoding.hpp"
+namespace cs {
+
+    void TargetMetricHandler::makeTargetMetric(const SynthesisData& data, const Configuration& configuration) {
+        bool useMaxSat = configuration.strategy == OptimizationStrategy::UseMinimizer || configuration.strategy == OptimizationStrategy::SplitIter;
+        bool onlyCnot  = configuration.target == TargetMetric::TWO_QUBIT_GATES;
+
+        switch (configuration.target) {
+            case TargetMetric::GATES:
+            case TargetMetric::TWO_QUBIT_GATES:
+                makeGateMetric(data, useMaxSat, onlyCnot);
+                break;
+            case TargetMetric::DEPTH:
+                makeDepthMetric(data, useMaxSat);
+                break;
+            case TargetMetric::FIDELITY:
+                makeFidelityMetric(data, useMaxSat, configuration.architecture);
+                break;
+        }
+    }
+    void TargetMetricHandler::makeGateMetric(const SynthesisData& data, bool useMaxSat, bool onlyCNOT) {
+        logicbase::LogicTerm changes = logicbase::LogicTerm(true);
+        // COST
+        if (useMaxSat) {
```

Comment:
this function does nothing if `useMaxSat == false`. The same holds for the other functions in this file.
Just remove the whole parameter from these functions and make the corresponding check once at the call site.
Safes duplication, unnecessary parameters and a level of indentation.